Cargo.toml

```diff
@@ -76,6 +76,12 @@ uuid = { version = "1.6.1", features = ["serde", "js", "v4", "v7"] }
 [target.'cfg(unix)'.dependencies]
 nix = "0.26.4"
 
+[target.'cfg(any(target_os = "linux", target_os = "macos", target_os = "ios"))'.dependencies]
+snmalloc-rs = "0.3.4"
```

Comment:
I'm really interested to find out if you've noticed any changes in memory usage after making this update and running the tests. 

I've been diving into different allocators for the KV store lately, and any insights you can share would be great

Reply:
One thing I would say - you can only have a single `#[global_allocator]` for a Rust project and any sub crates. So this is probably not something that should be specified in a library (as it will prevent any project which imports that crate from specifying their own global allocator).

However, having a recommendation for a global allocator setup (or having a secondary crate which sets up some custom allocators for the different operating systems, like in `src/mem/mod.rs`), would be a great addition!

Reply:
Ack! That looks like a significant improvement, I am guessing the memory consumtion was way more than that with the default allocator. I was using jemalloc only in the benchmarks to check for improvements but this helps